### PR TITLE
refactor: Organizes the Select files

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Organizes the `Select` files. No functional changes.
- Extracts types to a `types.ts` file
- Extracts styles to a `styles.tsx` file
- Extracts the async stories to its own file

@cccs-RyanK 

### TESTING INSTRUCTIONS
1 - Execute all tests
2 - All tests should pass

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Select/utils.tsx [hunk 1]
```diff
@@ -16,33 +16,22 @@
...
+import { StyledHelperText, StyledLoadingText, StyledSpin } from './styles';
+import { LabeledValue, RawValue, SelectOptionsType, V } from './types';
 
 const { Option } = AntdSelect;
 
+export const MAX_TAG_COUNT = 4;
+
+export const TOKEN_SEPARATORS = [',', '\n', '\t', ';'];
+
+export const EMPTY_OPTIONS: SelectOptionsType = [];
```
Comment (villebro) [inline superset-frontend/src/components/Select/utils.tsx:33]: Should these (and others) be placed in a `constants.ts`?

Comment (michael-s-molina) [reply superset-frontend/src/components/Select/utils.tsx:33]: Great idea

File: superset-frontend/src/components/Select/utils.tsx [hunk 2]
```diff
@@ -121,104 +65,7 @@ export function hasOption(
...
-    border-bottom: 1px solid ${theme.colors.grayscale.light2};
-  `}
-`;
-
-export const MAX_TAG_COUNT = 4;
-export const TOKEN_SEPARATORS = [',', '\n', '\t', ';'];
-export const EMPTY_OPTIONS: SelectOptionsType = [];
-
-export const DEFAULT_SORT_COMPARATOR = (
+export const defaultSortComparator = (
```
Comment (ktmud) [inline superset-frontend/src/components/Select/utils.tsx:68]: Why the change? I thought we want all caps for constants?

Comment (ktmud) [reply superset-frontend/src/components/Select/utils.tsx:68]: Kind of feel this needs to be in `constants.ts`, too.

**Review by ktmud — APPROVED** [review]
Nice cleanup!

File: superset-frontend/src/components/Select/utils.tsx (hunk 2 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/Select/utils.tsx:68]: I was in doubt if I should treat it as a constant or another utility function. I'll revert the change and add it to the constants file.
